packages/docusaurus-theme-classic/src/index.js

```diff
@@ -109,3 +110,76 @@ module.exports = function (context, options) {
     },
   };
 };
+
+const NavbarLinkSchema = yup.object().shape({
+  items: yup
+    .array()
+    .optional()
+    .of(yup.lazy(() => NavbarLinkSchema.default(undefined))),
+  to: yup.string(),
+  href: yup.string().url(),
+  prependBaseUrlToHref: yup.bool().default(true),
+  label: yup.string(),
+  position: yup.string().oneOf(['left', 'right']).default('left'),
+  activeBasePath: yup.string(),
+  activeBaseRegex: yup.string(),
+  className: yup.string(),
+  'aria-label': yup.string(),
+});
+
+const ThemeConfigSchema = yup
+  .object()
+  .shape({
+    disableDarkMode: yup.bool().default(false),
+    image: yup.string(),
+    announcementBar: yup
+      .object()
+      .shape({
+        id: yup.string(),
+        content: yup.string(),
+        backgroundColor: yup.string().default('#fff'),
+        textColor: yup.string().default('#000'),
+      })
+      .optional(),
+    navbar: yup.object().shape({
+      hideOnScroll: yup.bool().default(false),
+      links: yup.array().of(NavbarLinkSchema),
+      title: yup.string().required(),
+      logo: yup.object().shape({
+        alt: yup.string(),
+        src: yup.string().required(),
+        srcDark: yup
+          .string()
+          .when('src', (value, schema) => schema.default(value)),
+        href: yup.string(),
+        target: yup.string(),
+      }),
+    }),
+    footer: yup.object().shape({
+      style: yup.string().oneOf(['dark', 'light']).default('light'),
+      logo: yup.object().shape({
+        alt: yup.string(),
+        src: yup.string(),
+        href: yup.string(),
+      }),
+      copyright: yup.string(),
+      link: yup.array().of(
+        yup.object().shape({
+          title: yup.string().required(),
+          items: yup.array().of(
+            yup.object().shape({
+              to: yup.string(),
+              href: yup.string().url(),
+              html: yup.string(),
+              label: yup.string(),
+            }),
+          ),
+        }),
+      ),
+    }),
+  })
+  .defined();
+
+module.exports.validateThemeConfig = ({validate, themeConfig}) => {
+  return validate(ThemeConfigSchema, themeConfig);
+};
```

Comment:
this assigns the validateThemeConfig to the default exported plugin function.

I'm not very fan of this idea, but this is a bit subjective (api design choices).

I think we have 2 cases here:
- 1) compiled plugins (ie in this repo, the ones in TS), where we can use ES imports/exports
- 2) "raw" plugins (like this one, and maybe the ones the user will write), for which we can't use ES imports/exports, otherwise we wouldn't support older nodejs versions

What about exporting these static methods directly?

1) => `export function validateThemeConfig() {};`

2) => `exports.validateThemeConfig = function() {};`

It could even be possible to support both syntaxes 🤪  wonder the one we should recommend to the users